examples/httpx_client.py

```diff
@@ -23,22 +22,26 @@
 logger = logging.getLogger("client")
 
 
-class H3Transport(QuicConnectionProtocol, httpcore.AsyncHTTPTransport):
+class H3ResponseStream(httpx.AsyncByteStream):
+    def __init__(self, aiterator: AsyncIterator[bytes]):
+        self._aiterator = aiterator
+
+    async def __aiter__(self) -> AsyncIterator[bytes]:
+        async for part in self._aiterator:
+            yield part
+
+
+class H3Transport(QuicConnectionProtocol, httpx.AsyncBaseTransport):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
         self._http = H3Connection(self._quic)
         self._read_queue: Dict[int, Deque[H3Event]] = {}
         self._read_ready: Dict[int, asyncio.Event] = {}
 
-    async def handle_async_request(
-        self,
-        method: bytes,
-        url: Tuple[bytes, bytes, Optional[int], bytes],
-        headers: Headers = None,
-        stream: httpcore.AsyncByteStream = None,
-        extensions: dict = None,
-    ) -> Tuple[int, Headers, httpcore.AsyncByteStream, dict]:
+    async def handle_async_request(self, request: httpx.Request) -> httpx.Response:
+        assert isinstance(request.stream, httpx.AsyncByteStream)
```

Comment:
Aside: Interesting code smell that the `httpx` API is forcing you into here.

Could be a useful prompt that we could?? expose `request.sync_stream` and `request.async_stream` accessors. These would return the correct type, or raise `RuntimeError` if used in the wrong context.

Don't feel strongly enough about this to do anything about it right now, but... interesting thought.

Pointer to the same style within `https`... https://github.com/encode/httpx/blob/master/httpx/_client.py#L1716-L1724